app/api/teams.py

```diff
@@ -219,3 +221,219 @@ async def extend_team_trial(
+            raise HTTPException(
+                status_code=400,
+                detail="Cannot merge a team into itself"
+            )
+
+        # Check if source team has active product associations first
+        source_products = db.query(DBTeamProduct).filter(DBTeamProduct.team_id == source_team.id).all()
+        if source_products:
+            product_names = [product.product_id for product in source_products]
+            raise HTTPException(
+                status_code=400,
+                detail=f"Cannot merge team '{source_team.name}' - it has active product associations: {', '.join(product_names)}. Please remove product associations before merging."
+            )
+
+        # Get team keys and users (only if no product associations found)
+        source_keys = db.query(DBPrivateAIKey).filter(DBPrivateAIKey.team_id == source_team.id).all()
+        target_keys = db.query(DBPrivateAIKey).filter(DBPrivateAIKey.team_id == target_team.id).all()
+        source_users = db.query(DBUser).filter(DBUser.team_id == source_team.id).all()
+
+        # Check for conflicts
+        conflicts = _check_key_name_conflicts(target_keys, source_keys)
+
+        # Apply conflict resolution strategy
+        if conflicts:
+            if merge_request.conflict_resolution_strategy == "cancel":
+                return TeamMergeResponse(
+                    success=False,
+                    message=f"Merge cancelled due to {len(conflicts)} key name conflicts",
+                    conflicts_resolved=conflicts,
+                    keys_migrated=0,
+                    users_migrated=0
+                )
+
+            source_keys = await _resolve_key_conflicts(
+                conflicts,
+                merge_request.conflict_resolution_strategy,
+                source_keys,
+                merge_request.rename_suffix,
+                db,
+                current_user
+            )
+
+        # Store team names before deletion
+        source_team_name = source_team.name
+        target_team_name = target_team.name
+
+        # Migrate users and keys
+        users_migrated = await _migrate_users_to_team(source_users, target_team.id, db)
+        keys_migrated = await _migrate_keys_to_team(source_keys, target_team.id, db)
+
+        # Update LiteLLM key associations
+        # Create a map of keys by region to avoid unnecessary DB queries
+        keys_by_region = {}
+        for key in source_keys:
+            if key.region_id not in keys_by_region:
+                keys_by_region[key.region_id] = []
+            keys_by_region[key.region_id].append(key)
+
+        # Update LiteLLM key associations for each region
+        for region_id, region_keys in keys_by_region.items():
+            # Get region info
+            region = db.query(DBRegion).filter(
+                DBRegion.id == region_id,
+                DBRegion.is_active == True
+            ).first()
+
+            if not region:
+                continue
```

Comment:
when would this be the case? and is it a problem if there are relevant keys associated with an inactive region?